### PR TITLE
Commited after resolving conflicts in the 45002_signup branch

### PR DESCRIPTION
**Pull request by gittiggit**

### CODE REVIEW

File: app/assets/stylesheets/users.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
```
Comment (chuanhtuan) [inline app/assets/stylesheets/users.scss]: remove unused file

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      flash[:success] = "Welcome to the Sample App!"
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: i18n

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      redirect_to @user
+    else
+      flash.now[:danger] = "ভাই ভাই ওই ভাই, ঠিক ঠাক সাইনাপ করুন"
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: i18n

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: remove ()

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      flash[:success] = "Welcome to the Sample App!"
+      redirect_to @user
+    else
+      flash.now[:danger] = "ভাই ভাই ওই ভাই, ঠিক ঠাক সাইনাপ করুন"
+      render 'new'
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: ()

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -11,8 +11,12 @@
 <body>
 <%= render 'layouts/header' %>
 <div class="container">
+  <% flash.each do |message_type, message| %>
+      <div class="alert alert-<%= message_type %>"><%= message %></div>
+  <% end %>
   <%= yield %>
   <%= render 'layouts/footer' %>
+  <%= debug(params) if Rails.env.development? %>
```
Comment (chuanhtuan) [inline app/views/layouts/application.html.erb]: remove

File: app/views/static_pages/home.html.erb [hunk 7]
```diff
@@ -6,7 +6,7 @@
     If you are developer please make it quickly. If you are user please stay with us.
   </h2>
 
-  <%= link_to "Sign up now!", '#', class: "btn btn-lg btn-primary" %>
+  <%= link_to "Sign up now!", signup_path, class: "btn btn-lg btn-primary" %>
```
Comment (chuanhtuan) [inline app/views/static_pages/home.html.erb]: i18n

File: app/views/users/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,24 @@
+<% provide(:title, 'Sign up') %>
+<h1>Sign up</h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(@user) do |f| %>
```
Comment (chuanhtuan) [inline app/views/users/new.html.erb]: ()

File: app/views/users/new.html.erb [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+        <%= f.label :email %>
+        <%= f.email_field :email %>
+
+        <%= f.label :password %>
+        <%= f.password_field :password %>
+
+        <%= f.label :password_confirmation, "Confirmation" %>
+        <%= f.password_field :password_confirmation %>
+
+        <%= f.submit "Create my account", class: "btn btn-primary" %>
```
Comment (chuanhtuan) [inline app/views/users/new.html.erb]: i18n

File: app/views/layouts/application.html.erb (hunk 6 above)
Comment (gittiggit) [inline app/views/layouts/application.html.erb]: We put it for development purpose. We will remove it in final version.

File: app/models/user.rb [hunk 10]
```diff
@@ -1,2 +1,6 @@
 class User < ActiveRecord::Base
+  validates :name, presence: true, length: {maximum: 50}
+  validates :email, presence: true, length: {maximum: 255}, format: {with: /\A[\w\d\-.]+@[a-z\d\-.]+\.[a-z]+\z/i}, uniqueness: {case_sensitive: false}
```
Comment (chuanhtuan) [inline app/models/user.rb]: 80 characters/line

**Pull request merged**